### ISSUE TITLE
Documentation for PnP Core with C#

### ISSUE THREAD

**Issue by brittanyru** [pnp/pnpcore#282]
I am trying to use the PnP package "PnP.Framework (0.1.94-preview)", but I am not finding any resources and documentation on how to use it. Is there any documentation for the APIs and how to use it? I am reading different articles here and there talking about registering azure application sometimes, and other times authenticating differently, but I don't find an official documentation on how to access a SharePoint site with these tools and how to do basic operations. Any guidance?

**Comment by jansenbe** [comment]
For a listitem the fields (I assume that's what you need) are handled slightly different as each list can have it's own fields and thus each list has a different schema.  When loading a list to work with items check https://pnp.github.io/pnpcore/using-the-sdk/listitems-intro.html#reading-list-items to get started. 

If you've an item loaded you also have the Values collection loaded which contains all the list field values:

```csharp
if (addedItem["Title"].ToString() == "abc") 
{
  // do something
}
```

In your case you're interested in the list item linked to a file, see https://pnp.github.io/pnpcore/using-the-sdk/files-intro.html#getting-file-properties for a detailed explanation on how or check below snippet uses the document path to load it in one go:

```csharp
string documentUrl = $"{context.Uri.PathAndQuery}/mylibrary/document.docx";
IFile testDocument = await context.Web.GetFileByServerRelativeUrlAsync(documentUrl, w => w.Author, w => w.ModifiedBy, w => w.ListItemAllFields);

if (testDocument.ListItemAllFields["MyField"].ToString() == "abc")
{
    // do something
}
```

If you want to go via the list id of the list item connected to the file this works:

```csharp
IList list = await context.Web.Lists.GetByTitleAsync("mylibrary", p => p.Items);
IListItem testDocumentListItem = await list.Items.GetByIdAsync(1);

if (testDocumentListItem["MyField"].ToString() == "abc")
{
    // do something
}
```

If you want to go via file id you can do something like this:

```csharp
IList list = await context.Web.Lists.GetByTitleAsync("graph", p => p.RootFolder);
var uniqueId = new Guid("e2a52975-ceae-41b3-9271-315a5cfba6b2");
IFile testDocument = await list.RootFolder.Files.GetFirstOrDefaultAsync(p => p.UniqueId == uniqueId,  w => w.Author, w => w.ModifiedBy, w => w.ListItemAllFields);

if (testDocument.ListItemAllFields["MyField"].ToString() == "abc")
{
    // do something
}
```

**Comment by jansenbe** [comment]
See also https://pnp.github.io/pnpcore/using-the-sdk/basics-getdata.html for new documentation on getting data

**Comment by jansenbe** [comment]
Hey @brittanyru : did above get you unblocked? If so feel free to close the issue.

**Comment by jansenbe** [comment]
Closing this one, if you've other PnP Core SDK questions feel free to ask them via a new issue
